scripts/check_e2e_tests_are_captured_in_ci_test.py

```diff
@@ -0,0 +1,358 @@
+
+    def test_read_protractor_file(self):
+        protractor_config_file = os.path.join(
+            DUMMY_CONF_FILES, 'dummy_protractor.conf.js')
+
+        protractor_config_file_swap = self.swap(
+            check_e2e_tests_are_captured_in_ci, 'PROTRACTOR_CONF_FILE_PATH',
+            protractor_config_file)
+        with protractor_config_file_swap:
+            actual_protractor_config_file = (
+                check_e2e_tests_are_captured_in_ci.read_protractor_conf_file())
+        self.assertEqual(
+            DUMMY_PROTRACTOR_CONF_FILE, actual_protractor_config_file)
+
+    def test_get_e2e_suite_names_from_jobs_travis_yml_file(self):
+        def mock_read_travis_yml_file():
+            travis_ci_file = python_utils.open_file(
+                os.path.join(
+                    DUMMY_CONF_FILES, '.dummy_travis.yml'), 'r').read()
+            travis_ci_dict = utils.dict_from_yaml(travis_ci_file)
+            return travis_ci_dict
+
+        dummy_path = self.swap(
+            check_e2e_tests_are_captured_in_ci,
+            'read_and_parse_travis_yml_file', mock_read_travis_yml_file)
+        with dummy_path:
+            actual_travis_jobs = (
+                check_e2e_tests_are_captured_in_ci
+                .get_e2e_suite_names_from_jobs_travis_yml_file())
+        self.assertEqual(DUMMY_TEST_SUITES, actual_travis_jobs)
+
+    def test_get_e2e_suite_names_from_script_travis_yml_file(self):
+        def mock_read_travis_yml_file():
+            travis_ci_file = python_utils.open_file(
+                os.path.join(
+                    DUMMY_CONF_FILES, '.dummy_travis.yml'), 'r').read()
+            travis_ci_dict = utils.dict_from_yaml(travis_ci_file)
+            return travis_ci_dict
+
+        dummy_path = self.swap(
+            check_e2e_tests_are_captured_in_ci,
+            'read_and_parse_travis_yml_file', mock_read_travis_yml_file)
+        with dummy_path:
+            actual_travis_script = (
+                check_e2e_tests_are_captured_in_ci
+                .get_e2e_suite_names_from_script_travis_yml_file())
+        self.assertEqual(DUMMY_TEST_SUITES, actual_travis_script)
+
+    def test_get_e2e_suite_names_from_script_travis_yml_file_fail(self):
+        def mock_read_travis_yml_file_fail():
+            travis_ci_file = python_utils.open_file(
+                os.path.join(
+                    DUMMY_CONF_FILES, '.travis_has_extra_test.yml'), 'r').read()
+            travis_ci_dict = utils.dict_from_yaml(travis_ci_file)
+            return travis_ci_dict
+
+        dummy_path = self.swap(
+            check_e2e_tests_are_captured_in_ci,
+            'read_and_parse_travis_yml_file',
+            mock_read_travis_yml_file_fail)
+        with dummy_path:
+            actual_travis_script = (
+                check_e2e_tests_are_captured_in_ci
+                .get_e2e_suite_names_from_jobs_travis_yml_file())
+        self.assertNotEqual(DUMMY_TEST_SUITES, actual_travis_script)
```

Comment:
What is the point of this test? It seems to be checking whether reading from a different file gives the wrong result, which of course it will.

If you want to do failure tests, they should be testing for a specific exception message / error, and there should be a positive assert for that exception message / error.

Reply:
Yes, you are right, this doesn't make sense, removed. 
Done.